### PR TITLE
[ENH] Clarify SEEG tutorial coordinate frames

### PR DESCRIPTION
**Pull request by alexrockhill**
This clarifies and I think simplifies what's going on with the coordinate transforms for the SEEG case.

I think this will be really helpful for the GUI as they will hopefully be able to follow the same steps.

### CODE REVIEW

**Comment by alexrockhill** [comment]
I need to add a test for the new method, doing that now

**Comment by alexrockhill** [comment]
Not to toot my own horn but I really like the applying a `trans` to a `montage`. I think it's very clean because the montage checks the coordinate frames for you, you just have to operate at the abstract level of coordinate frames, and it solves this issue that all `info['dig']` is stored in the "head" coordinate frame by just requiring a `trans` file and then the montage can be got and then transformed in just a few lines.

I think this works really well for finding channel coordinates from just a CT; the fiducials could be found on the CT and a head to CT transform could be used to get the coordinates in CT space and be applied directly to the montage.

**Review by agramfort — APPROVED** [review]
really nice !

**Comment by drammock** [comment]
`changes/latest.inc:118: WARNING: py:meth reference target not found: mne.channels.montage.apply_trans`

you need to add it to `doc/python_reference.rst` (or rather, one of the other `.rst` files in that directory that it includes)

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -115,6 +115,8 @@ Enhancements
 
 - Add :func:`mne.viz.Brain.add_volume_labels` to plot subcortical surfaces and other regions of interest (:gh:`9540` by `Alex Rockhill`_ and `Eric Larson`_)
 
+- Add :meth:`mne.channels.montage.apply_trans` to apply a transform directly to a montage (:gh:`9601` by `Alex Rockhill`_)
```
Comment (larsoner) [inline doc/changes/latest.inc:118]:
```suggestion
- Add :meth:`mne.channels.DigMontage.apply_trans` to apply a transform directly to a montage (:gh:`9601` by `Alex Rockhill`_)
```

**Review by larsoner — APPROVED** [review]

**Review by drammock — COMMENTED** [review]
I'm having trouble following `10_ieeg_localize.html`, although the places that I find difficult to follow (or where I find typos/awkward passages) appear to have not changed in this PR, so maybe this is not the best place to discuss fixes. @alexrockhill  LMK if you would prefer to make those changes in this PR, and if so, we can schedule a time to talk it through.

**Comment by alexrockhill** [comment]
> I'm having trouble following `10_ieeg_localize.html`, although the places that I find difficult to follow (or where I find typos/awkward passages) appear to have not changed in this PR, so maybe this is not the best place to discuss fixes. @alexrockhill LMK if you would prefer to make those changes in this PR, and if so, we can schedule a time to talk it through.

Sure that's fine, can you talk this morning? Feel free to coordinate on Discord

**Comment by larsoner** [comment]
From discord with @alexrockhill and @drammock we decided to merge this and @alexrockhill will 1) make a PR to clarify the narrative after the GUI work is done (or as part of it) and 2) open a dipy issue to see if SDR provides a direct (non-discretized) mapping from points in the moving space to the static space (i.e., the equivalent of an affine `np.dot(affine, np.r_[pt, 1.])[:3]`)

**Comment by larsoner** [comment]
Thanks @alexrockhill !

File: tutorials/clinical/10_ieeg_localize.py [hunk 2]
```diff
@@ -203,44 +202,28 @@ def plot_overlay(image, compare, title, thresh=None):
...
-elec_df = pd.read_csv(op.join(misc_path, 'seeg', 'sample_seeg_electrodes.tsv'),
-                      sep='\t', header=0, index_col=None)
-ch_names = elec_df['name'].tolist()
-ch_coords = elec_df[['R', 'A', 'S']].to_numpy(dtype=float)
+# MNE stores digitization montages in a coordinate frame called "head"
+# thus even though the channel positions were found in the reference frame of
+# Freesurfer surfaces when they were assigned to the ``raw`` object, they were
+# transformed to "head" so we need to have a transform to put them back
+subj_trans = mne.read_trans(op.join(misc_path, 'seeg',
+                                    'sample_seeg_trans.fif'))
```
Comment (jasmainak) [inline tutorials/clinical/10_ieeg_localize.py:213]: sorry I saw this late. Is it kosher to have sEEG data as fif files and thus be transformed to "head" coordinates? Don't you normally get the coordinates in MRI coordinates as a separate text file or mat file?

Comment (alexrockhill) [reply tutorials/clinical/10_ieeg_localize.py:213]: Yeah I agree it's not ideal but I think to be backward compatible with MEG/EEG, FIF requires it to be in the "head" frame. I've thought it a bit like how Freesurfer uses surface RAS as its internal representation frame, MNE uses "head". They are all arbitrary reference frames and at least there is a standard and you know how it's stored. Ideally the coordinate frame could be specified in the FIF but that seems like pretty large change.

**Pull request merged**